tests/scale_test/query_bundles_by_project_id.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+import json
+import logging
+from datetime import datetime
+from utils import utils
+
+
+def format_query(project_uuid, workflow_name):
+    es_query_path = utils.SUBSCRIPTION_QUERIES.get(workflow_name)
+    if not es_query_path:
+        raise ValueError(f'No subscription query available for {workflow_name}.')
+    query = utils.load_es_query(es_query_path)
+    project_id_query = {'match': {'files.project_json.provenance.document_id': project_uuid}}
+    must_match = query['query']['bool']['must']
+    must_match.append(project_id_query)
+    query['query']['bool']['must'] = must_match
+    return {"es_query": query}
+
+def get_dss_url(env):
+    if env == 'prod':
+        return 'https://dss.data.humancellatlas.org'
+    return f'https://dss.{env}.data.humancellatlas.org'
+
+def get_bundle_datetime(bundle_version):
+    return datetime.strptime(bundle_version, '%Y-%m-%dT%H%M%S.%fZ')
+
+def get_latest_bundle_versions(bundle_list):
+    bundle_ids = [b['bundle_uuid'] for b in bundle_list]
+    if len(bundle_ids) == len(set(bundle_ids)):
+        return bundle_list
+    else:
+        bundle_map = {}
```

Comment:
I might just have too much clojure/fp on the brain, but I might do something like this here: 
```suggestion
from itertools import groupby
from functools import reduce

        ...
        keyfn = lambda x : x['bundle_uuid']
        sorted_bundles = sorted(bundle_list, key=keyfn)
        grouped_bundles = groupby(sorted_bundles, keyfn)
        return [reduce(choose_more_recent_bundle, g) for k, g in grouped_bundles]
```
where `choose_more_recent_bundle` would replace `get_bundle_datetime`

Reply:
Thanks for the suggestion! It's much more elegant 😃